examples/workflow/collect-events.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Context,
+  StartEvent,
+  StopEvent,
+  Workflow,
+  WorkflowEvent,
+} from "@llamaindex/core/workflow";
+import { OpenAI } from "llamaindex";
+
+// Create LLM instance
+const llm = new OpenAI();
+
+// Create custom event types
+export class JokeEvent extends WorkflowEvent<{ joke: string }> {}
```

Comment:
I was first not a big fan of creating the events via subclassing and preferred a factory.

You can compare both options by checking this commit: https://github.com/run-llama/LlamaIndexTS/pull/1188/commits/0237d9563049140ee9b5b23de5c15f7924736e7e

Advantages of sub-classing:
1. No issue in dealing with unnamed events
2. Just one line per event definition (in TS)
3. Simpler events code in framework

Disadvantages of sub-classing:
1. Have to use the `data` property

So now I am preferring the sub-classing variant.